### PR TITLE
Sequence batch via http

### PR DESCRIPTION
**Pull request by lxning**
## Description

Please read our [CONTRIBUTING.md](https://github.com/pytorch/serve/blob/master/CONTRIBUTING.md) prior to creating your first pull request.

Please include a summary of the feature or issue being fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #(issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing

Please describe the Unit or Integration tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

pytest test_example_stateful_http.py
======================================= test session starts =======================================
platform linux -- Python 3.10.14, pytest-7.3.1, pluggy-1.4.0
rootdir: /home/ubuntu/serve
plugins: cov-4.1.0, mock-3.12.0
collected 1 item

test_example_stateful_http.py .                                                             [100%]

======================================== 1 passed in 5.37s ========================================

## Checklist:

- [ ] Did you have fun?
- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: test/pytest/test_example_stateful_http.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        t0.start()
+        t1.start()
+
+        t0.join()
+        t1.join()
+    finally:
+        test_utils.unregister_model(model_name)
+
+        # Clean up files
+        shutil.rmtree(Path(model_store) / model_name)
```
[the remark this answers is not in the document]
Comment (lxning) [reply test/pytest/test_example_stateful_http.py:142]: line 100 copy mar_file_path to model_store/model_name.

File: test/pytest/test_example_stateful_http.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    shutil.copytree(mar_file_path, Path(model_store) / model_name)
+
+    params = (
+        ("model_name", model_name),
+        ("url", Path(model_store) / model_name),
+        ("initial_workers", "2"),
+        ("synchronous", "true"),
+    )
+
+    test_utils.start_torchserve(
```
[the remark this answers is not in the document]
Comment (lxning) [reply test/pytest/test_example_stateful_http.py:109]: The default conftest does not allow to install dependencies.

[the remark this answers is not in the document]
Comment (mreso) [reply test/pytest/test_example_stateful_http.py:109]: Makes sense

File: frontend/server/src/main/java/org/pytorch/serve/util/ConfigManager.java [hunk 3]
```diff
@@ -960,6 +971,34 @@ public Double getTimeToExpiration() {
         return 0.0;
     }
 
+    public String getTsHeaderKeySequenceId() {
```
[the remark this answers is not in the document]
Comment (lxning) [reply frontend/server/src/main/java/org/pytorch/serve/util/ConfigManager.java:974]: This function is to get the key name for the sequence id in headers.

File: test/pytest/test_example_stateful_http.py [hunk 4]
```diff
@@ -126,7 +126,7 @@ def test_stateful_mar(mar_file_path, model_store):
             args=(
                 model_name,
                 "seq_1",
-                "1 3 6",
+                "1 3 6 10 15",
```
Comment (mreso) [inline test/pytest/test_example_stateful_http.py:129]: Longer sequence is good, but the sequences should be different to make sure we have no cross talking between them. Also why would we expect 1 3 6 for seq_0 if we send 5 requests?

**Review by mreso — APPROVED** [review]
LGTM now, but please recheck your test (see comment)

**Comment by mreso** [comment]
Removing from queue so we can address the last issue with the test.

**Review by mreso — APPROVED** [review]
LGTM

**Pull request merged**